libs/components/layout/testing/src/modules/text-expand-repeater/text-expand-repeater-harness.ts

```diff
@@ -0,0 +1,78 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+import { SkyTextExpandRepeaterListStyleType } from '@skyux/layout';
+
+import { SkyTextExpandRepeaterHarnessFilters } from './text-expand-repeater-harness-filters';
+import { SkyTextExpandRepeaterItemHarness } from './text-expand-repeater-item-harness';
+
+/**
+ * Harness for interacting with a text expand repeater component in tests.
+ */
+export class SkyTextExpandRepeaterHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-text-expand-repeater';
+
+  #getExpandCollapseButton = this.locatorForOptional(
+    '.sky-text-expand-repeater-see-more',
+  );
+  #getOL = this.locatorForOptional('ol.sky-text-expand-repeater-container');
+  #getUL = this.locatorForOptional('ul.sky-text-expand-repeater-container');
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyTextExpandRepeaterHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyTextExpandRepeaterHarnessFilters,
+  ): HarnessPredicate<SkyTextExpandRepeaterHarness> {
+    return SkyTextExpandRepeaterHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Clicks the button element that expands or collapses text.
+   */
+  public async clickExpandCollapseButton(): Promise<void> {
+    const button = await this.#getExpandCollapseButton();
+
+    if (button === null) {
+      throw Error('Could not find button element.');
```

Comment:
could we be more descriptive here? The button wouldnt exist if the data isnt as long as the 'maxItems` value right? esp since we dont have design docs for this component i think that would be helpful here

Reply:
updated